### PR TITLE
Fix visual script icons

### LINKED ISSUE #50557: icons not representative of data
### Godot version

4.0.dev

### System information

Windows 10

### Issue description

In VisualScript the icons are out of oreder.
Vector2 Icon now is Plane, Transform now shows RID etc.
image

### Steps to reproduce

open attached VisualScript press rmb

### Minimal reproduction project

_No response_

### PR DESCRIPTION
**Pull request by SirQuartz**
This pull request fixes an issue where the visual script icons weren't representative of their data.

Fixes: #50557

node3d

### CODE REVIEW

File: editor/property_selector.cpp [hunk 1]
```diff
@@ -135,9 +135,9 @@ void PropertySelector::_update_search() {
 			search_options->get_theme_icon(SNAME("Quaternion"), SNAME("EditorIcons")),
 			search_options->get_theme_icon(SNAME("AABB"), SNAME("EditorIcons")),
 			search_options->get_theme_icon(SNAME("Basis"), SNAME("EditorIcons")),
+			search_options->get_theme_icon(SNAME("NodePath"), SNAME("EditorIcons")),
 			search_options->get_theme_icon(SNAME("Transform3D"), SNAME("EditorIcons")),
 			search_options->get_theme_icon(SNAME("Color"), SNAME("EditorIcons")),
-			search_options->get_theme_icon(SNAME("NodePath"), SNAME("EditorIcons")),
```
Comment (akien-mga) [inline editor/property_selector.cpp:140]: Why this move? The types should be kept in the order of `core/variant/variant.h`:
```
                // misc types
                COLOR,
                STRING_NAME,
                NODE_PATH,
                RID,
```

Not sure why `StringName` isn't included but the rest should be kept ordered.

Comment (SirQuartz) [reply editor/property_selector.cpp:140]: There was a problem with the way the icons appeared within the visual script based on their orderings here. In particular, Transform3D would should up where it should be path and vise versa, so by switching them this way they appeared as they should within the editor. Through trial and error, I found that having this arrangement seemed to resolve it. `StringName` was never there to begin with, not sure why it's missing 🤔

Comment (SirQuartz) [reply editor/property_selector.cpp:140]: There was a problem with the way the icons appeared within the visual script based on their orderings here. In particular, Transform3D would show up where it should be NodePath and vise versa, so by switching them this way they appeared as they should within the editor. Through trial and error, I found that having this arrangement seemed to resolve it. `StringName` was never there to begin with, not sure why it's missing 🤔

Comment (SirQuartz) [reply editor/property_selector.cpp:140]: Alright, I've tested re-ordering the types as they appear in `core/variant/variant.h` here's: what the Node3D looks like now:

node3d
This seems more accurate than before, though, I don't really know for sure.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**